r3179 | vapier | 2010-10-03 17:20:25 -0400 (Sun, 03 Oct 2010) | 1 line

fix infinite recursion with ip invocation #339013 by Krzysztof OlÄ™dzki

Index: net-scripts/net/iproute2.sh
===================================================================
--- lib/rcscripts/net/iproute2.sh (revision 3178)
+++ lib/rcscripts/net/iproute2.sh (revision 3179)
@@ -5,7 +5,8 @@
 # Fix any potential localisation problems
 # Note that LC_ALL trumps LC_anything_else according to locale(7)
 ip() {
-	LC_ALL=C ip "$@"
+	# use the full path so we don't recurse into ip()
+	LC_ALL=C $(type -P ip) "$@"
 }
 iproute2_tunnel() {
 	LC_ALL=C ip tunnel "$@"
